agent/virus_total_agent.py

```diff
@@ -55,23 +59,46 @@ def process(self, message: msg.Message) -> None:
                 not in self.whitelist_types
             ):
                 return None
-            response = virustotal.scan_file_from_message(
-                file_content=file_content, api_key=self.api_key
-            )
-            self._process_response(response, message.data.get("path"))
+            self._scan_file(file_content, message.data.get("path"))
         else:
             targets = self._prepare_targets(message)
             for target in targets:
-                response = virustotal.scan_url_from_message(target, self.api_key)
-                self._process_response(response, target)
+                self._scan_url(target)
+
+    @tenacity.retry(
+        stop=tenacity.stop_after_attempt(NUMBER_RETRIES),
+        wait=tenacity.wait_fixed(WAIT_TIME),
+        retry=tenacity.retry_if_exception_type(virustotal.VirusTotalApiError),
+        retry_error_callback=lambda retry_state: None,
+    )
+    def _scan_file(self, file_content: bytes, target: str | None) -> None:
```

Comment:
Why the retry?

Reply:
Retry after a minute if we reach the rate limit instead of not scanning the target!

Reply:
why? The rate limit is 4 per minute we need just wait another minute.